### ISSUE TITLE
Possibility to mark program as "favorite" & have a separate list with all favorites

### ISSUE THREAD

**Issue by zynx001** [add-ons/plugin.video.viervijfzes#53]
It would be nice to have a list with favorites (like in the vrtnu and vtmgo addon).
The fact that vier.be does not offer such functionality probably makes this more difficult, but a locally saved list of favorites would perhaps be a solution?

**Comment by michaelarnauts** [comment]
This is definitely possible and isn't that hard. We could store a json in our addon_data folder with a list of the id's of the programs you follow. By using the full a-z listing, we can map these to a name, description and cover.

**Comment by dagwieers** [comment]
As a user of a MySQL-backed Kodi I would prefer something that keeps this list in sync between devices. We really ought to have a secure Kodi add-on backend service for synchronized settings and data 😉

**Comment by zynx001** [comment]
Having such data synced automatically would of course be even better, but that would require some online storage for the data. I assumed that that would be a bit much to ask for... :-)

**Comment by michaelarnauts** [comment]
> As a user of a MySQL-backed Kodi I would prefer something that keeps this list in sync between devices. We really ought to have a secure Kodi add-on backend service for synchronized settings and data

Let's hope telenet launches a better platform for VIER, VIJF and ZES in a few months that allows keeping a list of favourites.

I don't think there is such a solution to store add-on data in the database directly, unless add-on settings are synchronized between instances?

**Comment by dagwieers** [comment]
This is now possible with GoPlay !

**Comment by michaelarnauts** [comment]
The GoPlay website is using amazon cognito sync to store the list. We just need to find a way to integrate with that. We can do that directly (like the code of @basrieter that we use for the authentication), or we could depend on boto3 to do this. 

In case we want to go the boto3 way, we would need an additional dependency on boto3, but that isn't in the Kodi repo yet. I've prepared something here in case we'll need this: https://github.com/michaelarnauts/script.module.boto3

I prefer to do it ourself (so we don't need to depend on boto3), but it can be a bit of work to make it work reliably.

**Comment by wardwygaerts** [comment]
Thanks the implementation! Is working perfect

**Comment by zynx001** [comment]
Thanks for the changes !
I will update my add-on and give it a try :-)
